### PR TITLE
Improve danger checks

### PR DESCRIPTION
**Pull request by marianeum**
Made our danger check messages a bit clearer and added some missing ones.

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,6 +50,7 @@ captures/
 .idea/dictionaries/
 .idea/caches/
 .idea/codeStyles/
+.idea/copilot/
```
Comment (marianeum) [inline .gitignore:53]: this really wanted to be added

File: Dangerfile.df.kts [hunk 2]
```diff
@@ -37,7 +37,7 @@ danger(args) {
         )
         val packageFilesCreated = createdFiles.any { it.startsWith("Backpack/src/main/java") }
         if (packageFilesCreated && !usageDetector && !isTrivial) {
-            warn("One or more package files were created, but `BpkComponentUsageDetector.kt` wasn't updated.")
+            warn("One or more package files were created, but `BpkComponentUsageDetector.kt` wasn't updated. If your component is an equivalent of a material or core component please add it to the detector.")
```
Comment (marianeum) [inline Dangerfile.df.kts:40]: improved the messaging

File: Dangerfile.df.kts [hunk 3]
```diff
@@ -46,16 +46,28 @@ danger(args) {
...
         }
 
         // If any screenshots were created, README should"ve updated.
         val screenshotsCreated = createdFiles.any { it.startsWith("docs") && it.contains("/screenshots/") }
-        if (screenshotsCreated) {
-            val readmeUpdated = allSourceFiles.any { it.startsWith("docs") && it.endsWith("README.md") }
-            if (!readmeUpdated) {
-                warn("One or more screenshot created, but `README.md` wasn't updated. Please include screenshots in README")
-            }
+        val readmeUpdated = allSourceFiles.any { it.startsWith("docs") && it.endsWith("README.md") }
```
Comment (marianeum) [inline Dangerfile.df.kts:54]: just moved this out of the if statement as it'll be reussed for some other checks

File: Dangerfile.df.kts [hunk 4]
```diff
@@ -46,16 +46,28 @@ danger(args) {
...
+        if (screenshotsCreated && !readmeUpdated) {
+            warn("One or more screenshot created, but `README.md` wasn't updated. Please include screenshots in README")
+        }
+
+        // If a component was updated, the corresponding README, tests and screenshots should have been updated.
```
Comment (marianeum) [inline Dangerfile.df.kts:59]: added a few more checks - these might be a little noisy in some areas where no visual changes happen, but will be helpful in most cases where these changes are missed

**Review by alejandrorosas — APPROVED** [review]

**Pull request merged**